src/sentry_plugins/github/plugin.py

```diff
@@ -70,101 +73,84 @@ def get_link_existing_issue_fields(self, request, group, event, **kwargs):
             'required': False
         }]
 
+    def get_client(self, project, user):
+        auth = self.get_auth_for_user(user=user)
+        if auth is None:
+            raise PluginError(ERR_UNAUTHORIZED)
+        return GitHubClient(token=auth.tokens['access_token'])
+
     def handle_api_error(self, error):
-        msg = u'Error communicating with GitHub: %s' % error
         status = 400 if isinstance(error, PluginError) else 502
         return Response({
             'error_type': 'validation',
-            'errors': {'__all__': msg},
+            'errors': {'__all__': self.message_from_error(error)},
         }, status=status)
 
-    def get_allowed_assignees(self, request, group):
-        try:
-            _url = self.build_api_url(group, 'assignees')
-            req = self.make_api_request(request.user, _url)
-            body = safe_urlread(req)
-        except (requests.RequestException, PluginError) as e:
-            raise PluginError(u'Error communicating with GitHub: %s' % e)
+    def message_from_error(self, exc):
+        if isinstance(exc, ApiUnauthorized):
+            return ERR_UNAUTHORIZED
+        elif isinstance(exc, ApiError):
+            if exc.code == 404:
+                return ('GitHub returned a 404 Not Found error. If this repository exists, ensure you have access to it and that Sentry is being allowed to read data from this organization (Account Settings > Authorized Applications > Sentry).')
```

Comment:
I think we the line length checks disabled since the hard rule of X chars generally isn't good. That said this is crazy long so I fixed.